libopenage/renderer/opengl/shader_program.h

```diff
@@ -33,10 +33,16 @@ class GlShaderProgram final : public ShaderProgram, public GlSimpleObject {
 	/// and draws the Geometry if it's not nullptr.
 	void execute_with(const GlUniformInput*, const GlGeometry*);
 
+	void send_uniform(const GlUniformInput *unif_in);
+
 	bool has_uniform(const char*) override;
 
+	int texture_array();
```

Comment:
What do these functions do? There is no explanation (add comments please).

Reply:
The texture_array() function basically sets the uniforms of the textures in the shader. Its a little implementation specific right now, but we can abstract it. the main reason here is that I do not have to pass in texture objects to the shader to bind them. They are bound by their own bind() method.